physics.cc

```diff
@@ -103,22 +103,22 @@ PScene::~PScene() {
   abort();
 }
 
-unsigned int PScene::simulate(unsigned int *ids, float *positions, float *quaternions, float *scales, unsigned int numIds, float elapsedTime) {
+unsigned int PScene::simulate(unsigned int *ids, float *positions, float *quaternions, float *scales, unsigned int numIds, float elapsedTime, float *velocities) {
```

Comment:
I think this entire function is confused and has become useless.

It is making a lot of choices for the programmer now, like disabling gravity, ignoring arguments, etc. that it is clearly nothing more than a hack for our case.

We should stop doing this update injection in the simulate step, and every update case we need should be handled by a method we can call to set the property on the object (e.g. ). Then we use that instead of keeping this updates array.